### ISSUE TITLE
Question for toonification model

### ISSUE THREAD

**Issue by jis478** [eladrich/pixel2style2pixel#30]
Hi, thanks again for the great work. 

"We train the pSp encoder to directly reconstruct real face images inside the toons latent space resulting in a projection of each image to the closest toon."

Regarding the above statement, did you mean that your pSp encoder was trained with a dataset consisting of only FFHQ images, not Toonified images?  If so, were reconstructed images (Toonified by generator) and input images (FFHQ ) were used for calculating losses?

**Comment by yuval-alaluf** [comment]
Hi @jis478 
Yes, our encode is trained **only** on real FFHQ images. That is, no paired data was used.  
We compute the losses between the real FFHQ input image and the generated toonified image.

**Comment by jis478** [comment]
Hi @yuval-alaluf , once again thanks for the answer. 

I'm playing around with a different generator model (fine-tuned on FFHQ StyleGANv2 model with a cartoon dataset) but the encoder fails to generate meaningful latent codes. 

I assume that for the toonification case toonified ones still looked like human faces to some extent so those losses worked pretty good. However in my case generated images are more like cartoons, so the textures and fine features are hugely different from those of human faces.

**Comment by yuval-alaluf** [comment]
Regarding the cartoon training: since you're trying to train on a new dataset, you will probably need to play around with the loss lambdas to get better results. As you said, the cartoon dataset is quite different than the toonify-style images. Another possible option is to train using paired data (if available) which may lead to better results. 

Regarding the second question, I am not sure what you mean by "celebA to sketch" task. Do you mean sketch-to-image? If so, notice that we train the sketch-to-image using data pairs of (sketch, real image). Therefore the encoder does see sketches during training.

**Comment by jis478** [comment]
Hi @yuval-alaluf,

Thanks for the reply. I've generated paired datasets and followed your sketch-to-image task, and I was able to generate images with decent quality. I haven't tried modifying losses yet but definitely I'm interested in it as preparing paired datasets are a pain in the ass for many applications.

**Comment by yuval-alaluf** [comment]
I am quite curious as to how your cartoon dataset looks like that the sketch to image model gave you decent paired images 😄

**Comment by jis478** [comment]
Hi @yuval-alaluf, sorry for my late reply. This is one of examples generated by the encoder. I was quite surprised that the encoder was able to remove some artifacts given in the target image. 

![cap1](https://user-images.githubusercontent.com/19499513/99897116-2cbcde80-2cda-11eb-8b9f-4492527db280.JPG)
